### PR TITLE
auth_module : made API for registration, login, viewProfile, updatePr…

### PR DESCRIPTION
**Pull request by vidhirathod29**
…ofile

### CODE REVIEW

File: models/auth.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+      gender: {
+        type: Sequelize.STRING,
+        enum: [GENDER.MALE, GENDER.FEMALE],
+      },
+      profile_image: {
+        type: Sequelize.STRING,
+      },
+      role: {
+        type: Sequelize.STRING,
+        enum: [ROLES.ADMIN, ROLES.CUSTOMER],
```
Comment (ShrutiSachpara) [inline models/auth.js:34]: add vendor role

File: middleware/authentication.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
+const jwt = require('jsonwebtoken');
+const { GeneralError } = require('../utils/error');
+const { StatusCodes } = require('http-status-codes');
+const { RESPONSE_STATUS } = require('../utils/enum');
+const { Messages } = require('../utils/messages');
+const logger = require('../logger/logger');
+
+const generateToken = (req) => {
+  const token = jwt.sign(
+    { email: req.email, password: req.password, role: req.role, id: req.id },
```
Comment (prajapatiRgithub) [inline middleware/authentication.js]: Remove password here no need to that

File: models/auth.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
+const { ROLES, GENDER } = require('../utils/enum');
+
+module.exports = (sequelize, Sequelize) => {
+  return sequelize.define(
+    'users',
+    {
```
Comment (prajapatiRgithub) [inline models/auth.js:6]: Please mention which is required or not. in entire file.
and also added length as well

File: service/authService.js [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+          RESPONSE_STATUS.SUCCESS,
+        ),
+      );
+    }
+  } else {
+    logger.error(`User ${Messages.ALREADY_EXIST}`);
+    next(
+      new GeneralError(
+        `User ${Messages.ALREADY_EXIST}`,
+        StatusCodes.FORBIDDEN,
```
Comment (prajapatiRgithub) [inline service/authService.js]: Please mention appropriate status code.

File: service/authService.js [hunk 5]
```diff
@@ -0,0 +1,188 @@
...
+        Messages.INCORRECT_CREDENTIAL,
+        StatusCodes.UNAUTHORIZED,
+        undefined,
+        RESPONSE_STATUS.ERROR,
+      ),
+    );
+  } else {
+    let token = generateToken({
+      email,
+      password,
```
Comment (prajapatiRgithub) [inline service/authService.js]: please don't add password in token

**Review by SwetaaRampariya — APPROVED** [review]

**Review by prajapatiRgithub — APPROVED** [review]

**Review by prajapatiRgithub — APPROVED** [review]

**Pull request merged**